config/data_funcs.go

```diff
@@ -196,3 +198,39 @@ outer:
 	log.Debugf("Input obj %v", result)
 	return result
 }
+
+// GetCmdLineValues return a list of all values for a key (used for non rancher. boot cmdline)
+// OK, so the kernel boot cmdline is not unicode
+func GetCmdLineValues(cmdline, key string) []string {
```

Comment:
1. Mine uses robust lib handling args mostly (if not at all) linux shell way, plus adds missing top level cloud-config standardized ssh_authorized_keys and few other. i.e. without this it is not possible to pass ssh key another way except as in cloud-config file or in OS state/image (confirming the problem in my production environment). While such possibility exists in Core OS. This really helped me a lot to deploy and move easily Core OS and later Rancher OS (after I implemented above) between stateless diskless  physical server machines with encrypted (cryptsetup disks considering those machines just as pools of resourcres (RAM, CPU, storage), while image was booted via over internet via iPXE https. The only thing I needed to prepare on server - is small active bootable partition with isolinux and iPXE, https certificates. I made pull request of same code still for v0.4.5 but it was not accepted due massive code changes due of moving to next version/branche so far understood an explanation. IMO, I would just do not mix wrapped them as e.g. cloud-config.ssh_authorized_keys so that e.g rancher.password would become cloud-config.rancher.password, config.rancher.autologin etc. appropriately, so it would never mixed and messed up with native linux kernel args and would be much more consistent and self explanatory.
2. Another inconsistency and even rather bug I found that  console=ttyXX does not instruct Rancher OS to show login screen on serial console. It shows only autologined shell in case if rancher.autologin=ttyXX specified. But I apparently do not want autologin on my dedicated and/or virtual server located  in hosting provider datacenter :-). I.e. it looks like  console=ttyXX applied only to boot process messages/log.
3. Another weird thing that e.g root=/dev/sda1 is not respected by Rancher OS to use /dev/sda1 as state partition. While it is very logical, expected and intutive IMO, especiall for Core OS users  where it works BTW (and BTW, same for sshkey just like alias of  ssh_authorized_keys and cloud-config-url=http://example.com/cloud-config like alias of rancher.cloud_init.datasources=[url:http://example.com/cloud-config])